### ISSUE TITLE
is_subtype does not work for Literals

### ISSUE THREAD

**Issue by apirogov** [erezsh/runtype#15]
```python
>>> is_subtype(Literal[1], Literal[1,2])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/admin/.cache/pypoetry/virtualenvs/metador-core-XpPben8F-py3.8/lib/python3.8/site-packages/runtype/validation.py", line 25, in is_subtype
    return t1 <= t2
TypeError: '<=' not supported between instances of 'OneOf' and 'OneOf'
```

Expected: `True`

**Comment by erezsh** [comment]
Fixed in `master`.

**Comment by apirogov** [comment]
Great, thanks! Seems to work as expected now :)

**Comment by apirogov** [comment]
I played around some more and found a few bugs:

`is_subtype(Literal[1,2], Union[Literal[1], Literal[2]])` -> `False`
`is_subtype(Union[Literal[1], Literal[2]], Literal[1, 2])` -> `True`

Why? I expect these to be the same type.

Also these do not work:

```python
is_subtype(Optional[Literal[1]], Literal[None, 1])
is_subtype(Literal[None, 1], Optional[Literal[1]])
is_subtype(Literal[True], bool)
is_subtype(Literal[1,2,3], int)
is_subtype(Literal["a","b"], str)
is_subtype(Tuple[Literal[1,2,3], str], Tuple[int, str])
```
results in `TypeError: '<=' not supported between instances of 'OneOf' and 'PythonDataType'` and similar errors

**Comment by erezsh** [comment]
Thanks for presenting all these use cases!

I merged a fix to `master`, and I believe it should work now.

(there's a minor issue with python3.8 that I'm not yet sure how to solve, but it's unlikely that you'll run into it)

**Comment by apirogov** [comment]
@erezsh 

Thanks! I found another bug:

```python
is_subtype(List[str], Optional[Union[List[str]]])
```

results in:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/metador-core-AXME58BY-py3.8/lib/python3.8/site-packages/runtype/validation.py", line 25, in is_subtype
    return ct1 <= ct2
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/metador-core-AXME58BY-py3.8/lib/python3.8/site-packages/runtype/base_types.py", line 102, in __ge__
    return any(other <= t for t in self.types)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/metador-core-AXME58BY-py3.8/lib/python3.8/site-packages/runtype/base_types.py", line 102, in <genexpr>
    return any(other <= t for t in self.types)
TypeError: '<=' not supported between instances of 'SequenceType' and '_NoneType'
```

(expected: `True`)

**Comment by erezsh** [comment]
@apirogov It's working for me. Can you please check with the latest `master` version?

**Comment by apirogov** [comment]
@erezsh I'm using the latest commit from master `349db61` with Python 3.8, just re-installed it again in a fresh environment to make sure... Still I have the same problem :(

**Comment by erezsh** [comment]
Yeah, *I* wasn't testing on the latest commit. I apologize!

I pushed a fix to `master`, please give it a try.

**Comment by apirogov** [comment]
Thanks! Now it works. :+1:

Could you give a rough summary of the "minor 3.8 issue" you mentioned above? As 3.8 is my primary 'target' in by project, I'd like to assess how relevant it is to what I'm doing... (I'm using runtime type checking very intesely in my project, so it makes me worry a bit)

**Comment by erezsh** [comment]
Sure. The problem is that in 3.8 (and only 3.8), `Literal[1] == Literal[True]`

That is causing runtype's caching to confuse them based on which one has been used first, which might cause expressions like `is_subtype(typing.Literal[True], bool)` to return False.

I do think there might be a way to solve this, I just haven't given it a lot of thought yet.

**Comment by apirogov** [comment]
Could you please release a new version of the package with the current state? I currently depend on the unreleased fixes and get the version from git, and this causes some complications and troubles.

**Comment by erezsh** [comment]
Yes, thanks for the reminder! Just released 0.3.2. A new release was way overdue.

I'm closing this issue. If the `Literal[1] == Literal[True]` thing is still bothering you, feel free to open a new issue for it!
